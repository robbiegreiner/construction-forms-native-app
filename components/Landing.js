import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  View,
  TextInput,
  Button,
  Alert,
  Image
} from 'react-native';

import firebase from '../firebase';
import CreateAccount from './CreateAccount';
import PropTypes from 'prop-types';


export default class Landing extends Component<{}> {
  constructor(){
    super()
  }

  signOut() {
    this.props.setUser();
    firebase.auth().signOut();
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.header}>
          Construction Forms
        </Text>
        <Image source={require('../assets/logo.png')}
        style={styles.logo}/>
        <Text style={styles.user}>
          Welcome, {this.props.user}!
        </Text>
        <Text style={styles.title}>
          Choose your form to complete!
        </Text>
        <Button
          style={styles.formButton}
          onPress={() => this.props.setView('hotwork')}
          title="HotWork Permit"
        />
        <Button
          style={styles.formButton}
          onPress={() => this.props.setView('pretask')}
          title="Pretask Plan Form"
        />
        <Button
          style={styles.formButton}
          onPress={() => this.signOut()}
          title="Sign Out"
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#E8C712',
    alignItems: 'center'
  },
  header: {
    marginTop: 100,
    fontSize: 36,
    textAlign: 'center',
  },
  logo: {
    width: 150,
    height: 150,
    marginTop: 20,
    marginBottom: 20
  },
  user : {
    textAlign: 'center',
    fontSize: 20,
    marginTop: 10,
    marginBottom: 30,
    fontWeight: 'bold'
  },
  title : {
    textAlign: 'center',
    marginBottom: 10,
  },
  formButton: {
    height: 50,
    width: 200,
    color: 'blue'
  }
});
