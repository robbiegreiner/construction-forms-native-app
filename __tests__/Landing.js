/* eslint-disable */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Landing from '../components/Landing';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, configure } from 'enzyme';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const tree = renderer.create(<Landing />);
  expect(tree).toBeDefined();
});

it('maches snapshot', () => {
  const tree = renderer.create(<Landing />).toJSON();
  expect(tree).toMatchSnapshot();
});
