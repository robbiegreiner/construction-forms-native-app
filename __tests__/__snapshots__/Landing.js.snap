// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E8C712",
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 36,
        "marginTop": 100,
        "textAlign": "center",
      }
    }
  >
    Construction Forms
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/logo.png",
      }
    }
    style={
      Object {
        "height": 150,
        "marginBottom": 20,
        "marginTop": 20,
        "width": 150,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 30,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome, 
    !
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Choose your form to complete!
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        HotWork Permit
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Pretask Plan Form
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Sign Out
      </Text>
    </View>
  </View>
</View>
`;
